modules/artifact/src/main/java/org/mule/module/artifact/classloader/ClassLoaderLookupPolicyFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.artifact.classloader;
+
+import org.mule.util.StringUtils;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *  Creates {@link ClassLoaderLookupPolicy} instances
+ */
+public class ClassLoaderLookupPolicyFactory
+{
+
+    private static String[] systemPackages = {
+            "java.",
+            "javax.",
+            "org.mule.",
+            "com.mulesoft.",
+    };
+
+    /**
+     * Creates a new lookup policy
+     *
+     * @param config comma separated list of overridden and blocked packages.
+     * @return a new lookup policy matching the provided config
+     * @throws IllegalArgumentException in case there is an attempt to override or block
+     * a system package.
+     */
+    public ClassLoaderLookupPolicy create(String config)
+    {
+        Set<String> overridden = new HashSet<>();
+        Set<String> blocked = new HashSet<>();
+
+        if (!StringUtils.isEmpty(config))
+        {
+            final String[] overrides = config.split(",");
+
+            if (overrides.length != 0)
+            {
+                for (String override : overrides)
+                {
+                    override = StringUtils.defaultString(override).trim();
+
+                    // 'blocked' package definitions come with a '-' prefix
+                    boolean isBlocked = override.startsWith("-");
+                    if (isBlocked)
+                    {
+                        override = override.substring(1);
+                    }
+
+                    String dottedOverride;
+                    if (override.endsWith("."))
```

Comment:
Use constants everywhere, especially when same string literal is used multiple times in same class.